Userland/Games/GameOfLife/BoardWidget.cpp

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright (c) 2021, Andres Crucitti <dasc495@gmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+#include "BoardWidget.h"
+#include <LibGUI/Painter.h>
+
+BoardWidget::BoardWidget(size_t rows, size_t columns)
+{
+    m_timer = add<Core::Timer>();
+    m_timer->stop();
+    m_timer->on_timeout = [this] {
+        run_generation();
+    };
+    m_timer->set_interval(m_running_timer_interval);
+
+    update_board(rows, columns);
+}
+
+void BoardWidget::run_generation()
+{
+    m_board->run_generation();
+    update();
+    if (m_board->is_stalled()) {
+        if (on_stall)
+            on_stall();
+        update();
+    };
+}
+
+void BoardWidget::update_board(size_t rows, size_t columns)
+{
+    set_running(false);
+
+    m_last_cell_toggled = columns * rows;
+
+    if (m_board) {
+        if (columns == m_board->columns() && rows == m_board->rows()) {
+            return;
+        }
+    }
+
+    m_board = make<Board>(rows, columns);
+}
+
+void BoardWidget::set_running_timer_interval(int interval)
+{
+    if (is_running())
+        return;
+
+    m_running_timer_interval = interval;
+    m_timer->set_interval(m_running_timer_interval);
+
+    if (on_running_state_change)
+        on_running_state_change();
+}
+
+void BoardWidget::set_running(bool r)
+{
+    if (r == m_running)
+        return;
+
+    m_running = r;
```

Comment:
Let's call this `running`:

```suggestion
void BoardWidget::set_running(bool running)
{
    if (running == m_running)
        return;

    m_running = running;
```